### ISSUE TITLE
Refactor fieldHolder to be applied via transforms instead of being the default export

### ISSUE THREAD

**Issue by NightJar** [silverstripe/silverstripe-admin#510]
This allows developers to decorate the component proper as opposed to the higher order component (fieldHolder).

As an example:
Recently trying to change `FieldGroup` via `Injector.transform` with another higher order component that returned a subclass of the FieldGroup (as is a reasonably common case with PHP Injector), I was met with an error for "function getClassName is undefined" - because instead of extending the FeildGroup as I was expecting, I was instead extending the HigherOrderComponent (fieldHolder).

This description is probably a bit convoluted, so I'll update the issue with the workaround addressing the above as a PR soon (as worked on with @unclecheese ).

**Comment by NightJar** [comment]
#511 solved the issue for a singular field, as a proof of concept for the PR (doing only what was necessary at the time instead of changing too much at once).
There are still [quite a few other fields](https://github.com/silverstripe/silverstripe-admin/blob/1/client/src/boot/registerComponents.js#L44-L61) that this separation of concerns can be applied to.

**Comment by raissanorth** [comment]
Yes please we are running into a similar issue as the one described above whilst working on https://github.com/dnadesign/silverstripe-elemental/issues/486
